pkg/utils/secrets/basic_auth.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secrets
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gardener/gardener/pkg/utils"
+	"k8s.io/apiserver/pkg/authentication/user"
+)
+
+type formatType string
+
+const (
+	// BasicAuthFormatNormal indicates that the data map should be rendered the normal way (dedicated keys for
+	// username and password.
+	BasicAuthFormatNormal formatType = "normal"
+	// BasicAuthFormatCSV indicates that the data map should be rendered in the CSV-format.
+	BasicAuthFormatCSV formatType = "csv"
+
+	// DataKeyCSV is the key in a secret data holding the CSV format of a secret.
+	DataKeyCSV = "basic_auth.csv"
+)
+
+// BasicAuthSecret contains the specification a to-be-generated basic authentication secret.
+type BasicAuthSecret struct {
+	Name   string
+	Format formatType
+
+	Username       string
+	PasswordLength int
+}
+
+// BasicAuth contains the username, the password and the format for serializing the basic authentication
+// secret.
+type BasicAuth struct {
+	Name   string
```

Comment:
Maybe use composition for these two values (i.e., Name and Format)  in a parent struct?